### DISCUSSION TITLE
Get list of existing layers

### DISCUSSION THREAD

**Discussion by AdrianMrn** [plouc/nivo#1969] [Ideas]
Is there a way to extend the existing list of layers on e.g. a BarChart instead of having to write out all the separate layers? Being able to do something like this would be nice:

```jsx
<Bar
  layers={(layers) => [...layers, MyCustomLayer]}
/>
```

Alternatively, being able to add a ref to the component where you can read out the layers would also work.

The reason I need this is because I'm passing in a custom layer that could go over different types of charts, so I don't necessarily have a list of the layers for that particular chart type on hand.

Let me know if you want me to open a PR for this.

**Comment by frederickf** [comment]
I know for the charts I've used the layers are included in the default props that are exported. Ex:

```
import { BarDefaultProps, Bar } from '@nivo/bar';

const layers = [
  ...BarDefaultProps.layers
];

```
